armi/reactor/tests/test_blocks.py

```diff
@@ -1704,6 +1704,40 @@ def test_getPinCenterFlatToFlat(self):
         f2f = self.HexBlock.getPinCenterFlatToFlat()
         self.assertAlmostEqual(pinCenterFlatToFlat, f2f)
 
+    def test_gridCreation(self):
+        b = self.HexBlock
+        # The block should have a spatial grid at construction,
+        # since it has mults = 1 or 169 from setup
+        b.autoCreateSpatialGrid()
+        self.assertTrue(b.spatialGrid is not None)
+        for c in b:
+            if c.getDimension("mult", cold=True) == 169:
+                # Then it's spatialLocator must be of size 217
+                locations = c.spatialLocator
+                self.assertEqual(type(locations), grids.MultiIndexLocation)
+                mult = 0
+                for loc in locations:
+                    mult = mult + 1
+                self.assertEqual(mult, 169)
+
+    def test_gridNotCreatedMultipleMultiplicities(self):
```

Comment:
Can we get something like this, but to test that we raise when the multiplicity for the pins doesn't fit snugly into the number of rings?